extras/log_file_server/bin/server.dart

```diff
@@ -0,0 +1,90 @@
+import 'dart:async' show Future;
+import 'dart:io';
+
+import 'package:shelf/shelf.dart';
+import 'package:shelf/shelf_io.dart';
+import 'package:shelf_router/shelf_router.dart';
+
+const rootDir = '/var/opt/oap/logs';
+
+// Configure routes.
+final _router = Router()
+  ..delete('/logs/deleteMe.log', _delete)
+  ..get('/logs/<path>', _get)
+  ..head('/logs/<path>', _head)
+  ..post('/logs/<path>', _post);
+
+Future<Response> _delete(Request req) async {
+  var file = File('$rootDir/deleteMe.log');
+  if (file.existsSync()) {
+    await file.delete();
+  }
+  return Response.ok(null);
+}
+
+Future<Response> _get(Request req, String path) async {
+  var file = File('$rootDir/$path');
+  if (!file.existsSync()) {
+    return Response.notFound(null);
+  }
+  var body = file.readAsStringSync();
+  return Response.ok(body);
+}
+
+Future<Response> _head(Request req, String path) async {
+  var file = File('$rootDir/$path');
+  if (!file.existsSync()) {
+    return Response.notFound(null);
+  }
+  var length = file.lengthSync();
+  return Response.ok(null, headers: {'content-length': '$length'});
+}
+
+Future<Response> _post(Request req, String path) async {
+  // validate received data
+  var mimeType = req.mimeType;
+  if (mimeType != 'text/plain') {
+    return Response.badRequest(body: 'Content-Type must be text/plain!');
+  }
+  if (req.isEmpty) {
+    return Response.badRequest(body: 'Body must not be empty!');
+  }
+
+  // get body
+  var length = req.contentLength;
+  var body = await req.readAsString();
+  if (length != body.length) {
+    return Response.badRequest(
+        body: 'Content-Length of $length '
+            'did not match body.length of ${body.length}!');
+  }
+
+  // // get remote address
+  // var connectionInfo =
+  //     req.context['shelf.io.connection_info'] as HttpConnectionInfo;
+  // var remoteAddress = connectionInfo.remoteAddress.address;
+  // print('remoteAddress = "$remoteAddress" (${remoteAddress.runtimeType})');
```

Comment:
These commented lines are for reading IP addresses?

Reply:
Yes. While it is best not to leave in commented-out code I thought I might use the remote IP address in some logs.